### PR TITLE
Resolve reset password confusion 

### PR DESCRIPTION
**Pull request by RawKStar77**
Meant to resolve https://github.com/hypothesis/h/issues/2140.

Text of email now reads:
```
Hello, {username}!

Someone requested resetting your password. If it was you, reset your password by using this reset code:
{code}

Alternatively, you can reset your password by clicking on this link:
{link}

If you don't want to change your password, please ignore this email message.

Regards,
{domain}
```

Try it for yourself here: http://reset-password-ux.dokku.hypothes.is/

### CODE REVIEW

File: h/accounts/views.py [hunk 1]
```diff
@@ -130,8 +135,60 @@ class AsyncAuthController(AuthController):
...
+        req = self.request
+        schema = req.registry.getUtility(IForgotPasswordSchema)
+        schema = schema().bind(request=req)
+
+        form = req.registry.getUtility(IForgotPasswordForm)
+        form = form(schema)
+
+        if req.method == 'GET':
+            if req.user:
+                return HTTPFound(location=self.forgot_password_redirect_view)
```
Comment (tilgovi) [inline h/accounts/views.py]: Travis informs us, via `make lint`, that this is not imported. Maybe change it to `httpexceptions.HTTPFound`.

File: h/accounts/views.py [hunk 2]
```diff
@@ -130,8 +135,60 @@ class AsyncAuthController(AuthController):
...
+        try:
+            captured = form.validate(controls)
+        except deform.ValidationFailure as e:
+            return {'form': e.render(), 'errors': e.error.children}
+
+        user = self.User.get_by_email(req, captured['email'])
+        activation = self.Activation()
+        self.db.add(activation)
+        user.activation = activation
+        Str = self.Str
```
Comment (tilgovi) [inline h/accounts/views.py]: PEP8 doesn't like this. I know it's comes from horus. Let's make it nicer, though. It's not used in enough places to bother with a local variable. Just use `self.Str` everywhere.

File: h/accounts/views.py [hunk 3]
```diff
@@ -130,8 +135,60 @@ class AsyncAuthController(AuthController):
...
+        body = emailtext.format(
+            code=user.activation.code,
+            link=route_url('reset_password', req, code=user.activation.code),
+            username=username, domain=req.application_url)
+        subject = Str.reset_password_email_subject
+        message = Message(subject=subject, recipients=[user.email], body=body)
+        mailer.send(message)
+
+        FlashMessage(self.request, Str.reset_password_email_sent,
+            kind='success')
```
Comment (tilgovi) [inline h/accounts/views.py]: More indentation needed.

**Comment by tilgovi** [comment]
Clean up the rest of `make lint` and this is okay.

**Comment by RawKStar77** [comment]
@tilgovi, lint approves now. :shipit:?

**Comment by tilgovi** [comment]
This is a straight copy/paste from horus, which simply doesn't allow us to customize the email text.

@nickstenning do we want to merge this? add tests first?

**Comment by tilgovi** [comment]
@judell can I get your eyes on the copy of the template? Should we mention the form they might have been presented with? Is the spacing good? Etc, etc.

**Comment by judell** [comment]
This?
https://github.com/hypothesis/h/pull/2158/files#diff-4b305d9ee5f5cb13a6dc9c877131c08fR168

**Comment by tilgovi** [comment]
Yes.

**Comment by judell** [comment]
https://gist.github.com/judell/b45a77f789dd78a6a527

Not sure what {domain} resolves to.

**Comment by RawKStar77** [comment]
> Not sure what {domain} resolves to.

https://hypothes.is

**Comment by judell** [comment]
@RawKStar77 thought so. @dwhly should we rather just sign as:

Regards,

The Hypothesis Team

or 

Hypothesis

or 

?

**Comment by dwhly** [comment]
I like team.

**Comment by tilgovi** [comment]
@RawKStar77 wanna clean that up and I'll merge this?

**Comment by RawKStar77** [comment]
@tilgovi I can easily put "The Hypothes.is Team" in the text of the email, but that's less nice for anyone who might fork the repository? Any ideas on how best to handle this? Should we go with "The Hypothes.is Team" for now? Or is it easy to add a variable somewhere?

**Comment by tilgovi** [comment]
He have other places in the repo already where we specifically say "Hypothesis". We will eventually need translation strings for these things and in the particular case of this once we separate ourselves from Horus more we should be templating all these emails in a way that's more extensible than they've done it.

But note the other details of Jon's suggestion: space above and below the code and link, for instance.

**Comment by RawKStar77** [comment]
> But note the other details of Jon's suggestion: space above and below the code and link, for instance.

Done. Anything else?

**Comment by tilgovi** [comment]
Removed the now unused "domain" argument, verified that this does what it says. I'm going to leave off writing tests for this until someone can take a comprehensive whack at improving this module.

**Comment by RawKStar77** [comment]
> Removed the now unused "domain" argument, verified that this does what it says.

Done.

selection_049

**Pull request merged**